src/uart/registers.rs

```diff
@@ -206,4 +206,17 @@ impl<const PORT_PTR: usize> ControlRegister<PORT_PTR> {
     set_parity_odd_even,
     /// # Check Parity Odd Even Select
     check_parity_odd_even}
+
+    bit_impl! {4, RW,
+    /// # Set Transmit Parity Generation Enable
+    set_transmit_parity_generation_enable,
+    /// # Check Transmit Parity Generation Enable
+    check_transmit_parity_genration_enable}
+
+    bit_impl! {0..=3, RW u16,
+
```

Comment:
When using `bit_impl!` for multiple bit operations, one should use the built-in type with the smallest size that can fit the bit range. Here you are using only **4** bits, but are trying to squeeze an entire `u16` into it. 

So for here:
```diff
- bit_impl! {0..=3, RW u16,
+ bit_impl! {0..=3, RW u8,
``` 

Also, not sure if this was intentional but there is an extra space here. 
```diff
bit_impl! {0..=3, RW u16, 
-
/// # Set Receive FIFO Threshold
```